ocs_ci/deployment/aws.py

```diff
@@ -436,6 +441,433 @@ def destroy_cluster(self, log_level="DEBUG"):
             verify_stack_deleted(stack_name)
 
 
+class AWSUPIRHELWORKERS(AWSUPI):
+    """
+    A class to handle AWS UPI with RHEL worker nodes
+    """
+    def __init__(self):
+        self.name = self.__class__.__name__
+        super(AWSUPIRHELWORKERS, self).__init__()
+        self.worker_vpc = None
+        self.worker_iam_role = None
+        self.worker_subnet = None
+        self.worker_security_group = None
+        self.worker_tag = None
+        self.cf = None
+        self.cluster_name = config.ENV_DATA['cluster_name']
+        # A dict for holding instance Name to instance object mapping
+        self.rhel_worker_list = {}
+        self.rhel_worker_user = constants.EC2_USER
+        self.client = boto3.client(
+            'ec2', region_name=config.ENV_DATA['region']
+        )
+
+    def deploy_cluster(self, log_cli_level='DEBUG'):
+        """
+        Deploy cluster overridden function from parent
+
+        Args:
+            log_cli_level (str): log level
+        """
+        if config.ENV_DATA['skip_ocp_deployment']:
+            super(AWSUPIRHELWORKERS, self).deploy_cluster(log_cli_level)
+        else:
+            prev_ocs_flag = config.ENV_DATA['skip_ocs_deployment']
+            # deploy only OCP
+            config.ENV_DATA['skip_ocs_deployment'] = 'true'
+            super(AWSUPIRHELWORKERS, self).deploy_cluster(log_cli_level)
+            self.add_rhel_workers()
+            config.ENV_DATA['skip_ocs_deployment'] = prev_ocs_flag
+            config.ENV_DATA['skip_ocp_deployment'] = 'true'
+            super(AWSUPIRHELWORKERS, self).deploy_cluster(log_cli_level)
+            config.ENV_DATA['skip_ocp_deployment'] = 'false'
+
+    def get_worker_resource_id(self, resource):
+        """
+        Get the resource ID
+
+        Args:
+            resource (dict): a dictionary of stack resource
+
+        Returns:
+            resource_id (str): ID of worker stack resource
+        """
+        return resource['StackResourceSummaries'][0]['PhysicalResourceId']
+
+    def gather_worker_data(self):
+        """
+        Gather various info like vpc, iam role, subnet,security group,
+        cluster tag from existing RHCOS workers
+        """
+        suffix = 'no0'
+        self.cf = boto3.client('cloudformation')
+        stack_name = f'{self.cluster_name}-{suffix}'
+        resource = self.cf.list_stack_resources(StackName=stack_name)
+        worker_id = self.get_worker_resource_id(resource)
+        ec2 = boto3.resource('ec2')
```

Comment:
Is there a reason we define this here instead of in the `__init__` like the ec2 client?

Reply:
Done, moved it inside init